packages/snapshot/src/client.ts

```diff
@@ -50,51 +51,53 @@ export interface SnapshotClientOptions {
 }
 
 export class SnapshotClient {
-  filepath?: string
-  name?: string
-  snapshotState: SnapshotState | undefined
   snapshotStateMap: Map<string, SnapshotState> = new Map()
 
   constructor(private options: SnapshotClientOptions = {}) {}
 
-  async startCurrentRun(
+  async setup(
     filepath: string,
-    name: string,
     options: SnapshotStateOptions,
   ): Promise<void> {
-    this.filepath = filepath
-    this.name = name
-
-    if (this.snapshotState?.testFilePath !== filepath) {
-      await this.finishCurrentRun()
-
-      if (!this.getSnapshotState(filepath)) {
-        this.snapshotStateMap.set(
-          filepath,
-          await SnapshotState.create(filepath, options),
-        )
-      }
-      this.snapshotState = this.getSnapshotState(filepath)
+    if (this.snapshotStateMap.has(filepath)) {
+      throw new Error('already setup')
     }
+    this.snapshotStateMap.set(
+      filepath,
+      await SnapshotState.create(filepath, options),
+    )
   }
 
-  getSnapshotState(filepath: string): SnapshotState {
-    return this.snapshotStateMap.get(filepath)!
+  async finish(filepath: string): Promise<SnapshotResult> {
+    const state = this.getSnapshotState(filepath)
+    const result = await state.pack()
+    this.snapshotStateMap.delete(filepath)
+    return result
+  }
+
+  skipTest(filepath: string, testName: string): void {
+    const state = this.getSnapshotState(filepath)
+    state.markSnapshotsAsCheckedForTest(testName)
   }
 
-  clearTest(): void {
-    this.filepath = undefined
-    this.name = undefined
+  clearTest(filepath: string, testId: string): void {
+    const state = this.getSnapshotState(filepath)
+    state.clearTest(testId)
   }
 
-  skipTestSnapshots(name: string): void {
-    this.snapshotState?.markSnapshotsAsCheckedForTest(name)
+  getSnapshotState(filepath: string): SnapshotState {
+    const state = this.snapshotStateMap.get(filepath)
+    if (!state) {
+      throw new Error('snapshot state not initialized')
```

Comment:
Can we maybe make it more presentable here, too? Since snapshot manager is a public API (even used in webdriverio, I think). Something like `The snapshot state for ${file} was not initialised. Did you call manager.setup()?`